src/taskFlow.ts

```diff
@@ -1,80 +1,91 @@
+// taskFlow.ts
 import TaskRegistry from './taskRegistry';
 import * as yaml from 'js-yaml';
 import * as fs from 'fs';
 import * as path from 'path';
 
-
 class TaskFlow {
-  static taskRegistry: TaskRegistry;
+    static taskRegistry: TaskRegistry;
 
-  static async configure(ymlPath: string) {
-    if (!this.taskRegistry) {
-      this.taskRegistry = new TaskRegistry();
-    }
+    static async configure(ymlPath: string) {
+        if (!this.taskRegistry) {
+            this.taskRegistry = new TaskRegistry();
+        }
 
-    await this.addTasksFromConfig(ymlPath);
-  }
+        await this.addTasksFromConfig(ymlPath);
+    }
 
-  static async execute(inputData: any) {
-    const matchedTask = this.taskRegistry.getMatchingTask(inputData);
+    static async execute(inputData: any) {
+        const matchedTask = this.taskRegistry.getMatchingTask(inputData);
 
-    if (matchedTask) {
-      // Use `await` to execute the task function
-      const result = await matchedTask(inputData);
-      // Handle the result if needed
-      return result;
-    } else {
-      console.error('No matching task found.');
-      return null;
-    }
-  }
+        if (matchedTask) {
+          if (typeof matchedTask === 'function') {
+              const taskFunction = matchedTask as (inputData: any) => Promise<any>;
+              const result = await taskFunction(inputData);
+              return result;
+          } else if (typeof matchedTask === 'function' || typeof matchedTask === 'object') {
+              const taskInstance = typeof matchedTask === 'function' ? (new (matchedTask as { new (): any })() as any) : matchedTask;
+              if (typeof taskInstance.condition === 'function') {
```

Comment:
Logic is fine, but suppose if we have 100 task and we execute it 100 times, then it will instantiate 100*100 times. So, instead can we add some code in registry to see if incoming task is of Class type then instantiate and add it in array instead of class itself.